source/simulator/core.cc

```diff
@@ -1678,19 +1678,38 @@ namespace aspect
             free_surface->execute ();
 
           assemble_advection_system (AdvectionField::temperature());
-          build_advection_preconditioner(AdvectionField::temperature(),
-                                         T_preconditioner);
-          solve_advection(AdvectionField::temperature());
+          if (system_rhs.block(AdvectionField::temperature().block_index(introspection)).l2_norm() <= std::numeric_limits<double>::min())
```

Comment:
Baa, what a stupid warning! Comparing against 0 should definitely be safe.

I guess in that case leave it as it is, and add a comment to the effect that we really just want to compare against zero but can't because of the warning...